### PR TITLE
add support for setting secret from env variable

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
add support for setting secret from env variable

**Related issue**
https://github.com/compose-spec/compose-spec/issues/251

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

**Comment by jmb-altana** [comment]
Thanks for the impl @ndeloof

File: pkg/compose/secrets.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	b := bytes.Buffer{}
+	tarWriter := tar.NewWriter(&b)
+	mode := uint32(0400)
+	if config.Mode != nil {
+		mode = *config.Mode
+	}
+
+	target := config.Target
+	if config.Target == "" {
+		target = "/run/secrets/" + config.Source
```
Comment (glours) [inline pkg/compose/secrets.go:67]: maybe we can use `path.Join` instead of plain string? (or `filepath.Join` if we need to target Windows containers)

**Review by glours — APPROVED** [review]
LGTM

**Review by laurazard — APPROVED** [review]
LGTM

**Pull request merged**